codegen/src/parser.rs

```diff
@@ -164,27 +166,41 @@ named!(message<Message>,
            msg
        }));
 
-named!(enum_field<(String, i32)>, 
+named!(hex_integer<i32>,
+    do_parse!(
+        tag!("0x") >>
+        num: map_res!(
+            map_res!(hex_digit,str::from_utf8),
+            |s| i32::from_str_radix(s,16)
+        ) >> (num)
+    ));
+
+named!(integer<i32>,  map_res!( map_res!(digit,str::from_utf8), str::FromStr::from_str));
+
+named!(enum_field<(String, i32)>,
        do_parse!(name: word >> many0!(br) >>
                  tag!("=") >> many0!(br) >>
-                 number: map_res!(map_res!(digit, str::from_utf8), str::FromStr::from_str) >> many0!(br) >>
+                 number: alt!(hex_integer | integer) >> many0!(br) >>
```

Comment:
The people I work with are EEs with C++ background. So they speak hex as their mother tongue, and tend to throw in random semi-colons!